apps/server/src/modules/fwu-learning-contents/uc/fwu-learning-contents.uc.ts

```diff
@@ -0,0 +1,41 @@
+import { Injectable, NotFoundException, InternalServerErrorException, Inject } from '@nestjs/common';
+import { GetObjectCommand, S3Client } from '@aws-sdk/client-s3';
+import { Readable } from 'stream';
+import { Logger } from '@src/core/logger';
+import { S3Config } from '../interface/config';
+
+@Injectable()
+export class FwuLearningContentsUc {
+	constructor(
+		private logger: Logger,
+		@Inject('S3_Client') readonly client: S3Client,
+		@Inject('S3_Config') readonly config: S3Config
+	) {
+		this.logger.setContext(FwuLearningContentsUc.name);
+	}
+
+	async get(path: string): Promise<Uint8Array> {
+		const request = new GetObjectCommand({
+			Bucket: this.config.bucket,
+			Key: path.toString(),
+		});
+		const response = await this.client.send(request).catch((error: unknown) => {
+			if (error && typeof error === 'object' && 'name' in error) {
+				if (error.name === 'NoSuchKey') {
+					throw new NotFoundException();
+				} else {
+					throw new InternalServerErrorException(error.name);
```

Comment:
may add additional logging here

Reply:
Can you specificate what you want?

Reply:
you mean { cause: error }

Reply:
Added additional information to NotFoundException

Reply:
@SteKrause please use in any of this 3 errors
throw new NotFoundException(undefined, { cause: errror});
throw new InternalServerErrorException(error.name, {cause: error});
throw new InternalServerErrorException(undefined, { cause: error});

Please check if undefined as input is allowed, otherwise use a message. If you do not use cause, we lost the origin error.

Reply:
Added:
throw new NotFoundException( { cause: errror.stack});
throw new InternalServerErrorException(error.name, {cause: error.stack});
throw new InternalServerErrorException(undefined, { cause: error});

Reply:
@Metauriel it's changed, but GitHub has some problems with the actions and workflows. Maybe it's not updated properly here in the comments.